### PR TITLE
Connecting a Frontend to a Backend Tutorial.

### PR DESCRIPTION
**Pull request by jeffmendoza**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1996)

### CODE REVIEW

File: docs/tutorials/connecting-apps/connecting-apps.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+* Route traffic to the backend using a frontend microservice.
+* Use Service objects to connect the frontend application to the
+  backend application.
+
+{% endcapture %}
+
+
+{% capture prerequisites %}
+
+* {% include task-tutorial-prereqs.md %}
```
Comment (steveperry-53) [inline docs/tutorials/connecting-apps/connecting-apps.md]: I think the prerequisites need to include using a cloud provider, like what I've done in https://github.com/kubernetes/kubernetes.github.io/pull/2006.

Comment (jeffmendoza) [inline docs/tutorials/connecting-apps/connecting-apps.md]: Same comment here as there, I hope we can have tutorials with `type: LoadBalancer` not require a cloud provider.

**Comment by jaredbhatti** [comment]
Ping, what's the status of this?

**Comment by jeffmendoza** [comment]
Ah, looks like were were deciding on `type: LoadBalancer` and if we need to list a cloud provider as a prereq. Minikube does work with `type: LoadBalancer`, but we probably want a page to link to

**Comment by chenopis** [comment]
@jeffmendoza @steveperry-53 Will that be addressed in this PR?

**Comment by jeffmendoza** [comment]
@steveperry-53 Rebased and fixed merge conflicts. Updated prereqs, but as we discussed we will need more work in this area for all tutorials.

@chenopis no, we still need more detailed explanation of `type: LoadBalancer` in relation to Tutorials, which environments support it, and how to use `type: NodePort` if your environment does not.

**Comment by steveperry-53** [comment]
@jeffmendoza, What's the next step for me? Are you ready for me to take a final look and then merge?

**Comment by jeffmendoza** [comment]
@steveperry-53 Yes, please.

**Comment by steveperry-53** [comment]
@jeffmendoza, See https://github.com/kubernetes/kubernetes.github.io/pull/1996/commits/1fdfd944176cfae05c382f356fa9e5cef149e19e for suggested edits.

Most of the edits are about these areas in our [style guide](https://kubernetes.io/docs/contribute/style-guide/): 1) present tense 2) not using "we" 3) code snippet format.

A few questions/comments:

1.  We should probably change the filename from connecting-apps.md to something like connect-frontend-backend.md.

2. The word "service" is overloaded in this topic. Could we use the word "application" unless we're talking about a Kubernetes Service? I like the way you've phrased this objective: "Use a Service object to connect the frontend application to the backend application."

3. Would it make sense to shorten the configuration files by removing fields that are incidental in this tutorial: resourceLimits, livenessProbe, etc?

**Comment by jeffmendoza** [comment]
Thanks for the fixes, they look great.

1. SG
2. Sounds good too, I'll do a pass, but I might need your help.
3. Yes, I'll trim those  (this config is from a more complete example).

**Comment by jeffmendoza** [comment]
@steveperry-53 updated.

**Comment by steveperry-53** [comment]
@jeffmendoza, If it were me, I'd eliminate the word "microservice" too. 

This tutorial shows how to connect a frontend application to a backend application by using a Kubernetes Service object.

* Use a Deployment to create and run a a backend application.
* Use a Deployment to create and run a frontend application.
* Use a Service object to connect the frontend to the backend.

I'll leave it up to you. I think we could publish this today. Would you like to make your final changes and squash?

**Comment by jeffmendoza** [comment]
Removing "microservice" is a little problematic as that is the point of having the Nginx frontend, theoretically it would reverse proxy based on URL to many "microservices". Generically we are connecting applications, that is the goal of the tutorial, but the example is a "microservice" setup.

**Comment by steveperry-53** [comment]
SGTM

**Comment by jeffmendoza** [comment]
@steveperry-53 squashed

**Comment by jeffmendoza** [comment]
Thanks Steve

**Pull request merged**